File: hgr/src/main/java/br/com/aplication/hgr/services/ProductService.java
package br.com.aplication.hgr.services;

import br.com.aplication.hgr.models.Product;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ProductService {

  Page<Product> listAllByPage(Pageable pageable  );

  Product findById( Long id );

  Product update( Product product );

  void save( Product product );

  Page<Product> getByDescription( Pageable pageable, String description );

  Page<Product> getByCode( Pageable pageable, String code );

  List<Product> getByProviderDocument( String providerDocument );
}

File: hgr/src/main/resources/static/js/app/services/productService.js
(function() {
  'use strict';

  angular.module('app')
    .factory('ProductService', ['$rootScope', '$location', '$http', '$resource',
      function($rootScope, $location, $http, $resource) {

        return $resource('product/:service/:id', {}, {
          save: {
            method: 'POST'
          },
          getById:{
            method: 'GET'
          },
          update:{
            method: 'PUT'
          },
          getByDescription: {
            method: 'GET',
            params: {
              service: 'description'
            }
          },
          getByCode: {
            method: 'GET',
            params: {
              service: 'code'
            }
          },
           getByProviderDocument: {
             method: 'GET',
             isArray: true,
             params: {
               service: 'providerDocument'
             }
           }
        });

      }
    ]);
})();